lib/scope.js

```diff
@@ -1200,96 +1234,65 @@ module.exports = {
       // TODO: Should we move this back into `getAllFields()`?
     }
 
-    // A lot of checkbox fields have a key hidden inside them. (not yesno type checkboxes)
-    let checkbox_match = layer_1_decoded.match( /^(.*)\[[BR]'(.*)'\]/ );
+    return scope.nestedDecode(scope, { to_handle: layer_1_decoded, field_like_names, nota });
+  },  // Ends scope.getPossibleVarNames()
+
+  nestedDecode: function ( scope, { to_handle, field_like_names = {}, nota } ) {
+    // A some fields have a key hidden inside them. (not yesno type checkboxes)
```

Comment:
I added param descriptions, and more comments in the function itself. I also removed some older, conflicting comments that weren't quite accurate with our new knowledge. Let me know if those help.

Reply:
Some great updates! It doesn't quite cover the different variations in the ways that I wanted, but maybe that can be made into its own issue. e.g.:

Here are the different encoding patterns DA creates for variable names:
- `foo` -> "encoded str"
- `foo[bar]` -> "encoded str" -> "foo[B'encoded str]" -> "foo[bar]"
- `foo[bar][baz][zoo]` -> "encoded str" -> "foo[B'encoded str]" -> "foo[B'encoded str]" - > "foo[bar][baz][zoo]"
- So on for the different patterns you found

Here are the different encoding patterns DA creates for values:
- Etc. Though maybe these are the same.

I think it's worth being comprehensive here because the details really make a difference for how we've decided to do this. I would have expected to need a loop, but we don't, and this shows the justification for that.

Again, this can be an issue for the future, referencing this comment: https://github.com/SuffolkLITLab/ALKiln/issues/574#issuecomment-1205562550. We can maybe use the interview you created in order to use legit encodings.

Reply:
I think pushing it off is the best idea, I don't quite have the time to do the due diligence on all of the variations right now. Making an issue of your most recent comment.